File: function-c++/palindrome.cpp
#include <iostream>
using namespace std;

// Fungsi untuk mengecek apakah kata adalah palincdrome
bool isPalindrome(const string &word)
{
    // Menghitung panjang kata
    int length = word.length();

    // Menginisialisasi indeks untuk karakter pertama dan terakhir
    int i, j;
    for (i = 0, j = length - 1; i < j; i++, j--)
    {
        // Membandingkan karakter dari kedua ujung
        if (word[i] != word[j])
        {
            return false; // bukan palindrome
        }
    }
    return true; // palindrome
}

int main()
{
    string kata;

    while (true)
    {
        // meminta pengguna untuk memasukan kata
        cout << "Masukan kata (q for exit): ";
        getline(cin, kata);
        // apabila kata q ditekan akan exit dari program
        if (kata == "q")
        {
            cout << "You exited the program" << endl;
            break;
        }
        // mengecek apakah kata adalah palindrome
        if (isPalindrome(kata))
        {
            cout << kata << " adalah palindrome" << endl;
        }
        else
        {
            cout << kata << " adalah bukan palindrome" << endl;
        }
    }
}

File: function-c++/palindrome.py
def is_palindrome(word):
    # Menghapus spasi dan mengubah huruf menjadi huruf kecil
    clean_word = word.lower().replace(" ", "")

    # Membandingkan kata dengan kebalikannya
    return clean_word == clean_word[::-1]

# To check the palindrome word
while True:
    kata_input = input("Masukan kata (q for exit): ")
    if kata_input == "q":
        print("You exited the program")
        break
    elif is_palindrome(kata_input):
        print(f"{kata_input} adalah palindrome")
    else:
        print(f"{kata_input} adalah bukan palindrome")